File: fluentcore/src/main/java/com/mcreater/fxfluent/tray/TrayManager.java
package com.mcreater.fxfluent.tray;

import java.awt.*;
import java.net.URL;

public class TrayManager {
    private static TrayIcon trayIcon;
    public static void init(URL url, String name, String des) throws AWTException {
        SystemTray tray = SystemTray.getSystemTray();

        Image image = Toolkit.getDefaultToolkit().createImage(url);

        trayIcon = new TrayIcon(image, name);
        trayIcon.setImageAutoSize(true);
        trayIcon.setToolTip(des);
        tray.add(trayIcon);
    }

    public static void display(String cap, String text, TrayIcon.MessageType type) {
        trayIcon.displayMessage(cap, text, type);
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/tray/TrayManager.kt
package com.mcreater.fxfluent.tray

import java.awt.AWTException
import java.awt.SystemTray
import java.awt.Toolkit
import java.awt.TrayIcon
import java.awt.TrayIcon.MessageType
import java.net.URL

class TrayManager {
    companion object {
        private var trayIcon: TrayIcon? = null
        @Throws(AWTException::class)
        fun init(url: URL?, name: String?, des: String?) {
            val tray = SystemTray.getSystemTray()
            val image = Toolkit.getDefaultToolkit().createImage(url)
            trayIcon = TrayIcon(image, name)
            trayIcon!!.setImageAutoSize(true)
            trayIcon!!.setToolTip(des)
            tray.add(trayIcon)
        }

        fun display(cap: String?, text: String?, type: MessageType?) {
            trayIcon!!.displayMessage(cap, text, type)
        }
    }
}